app/src/main/java/com/travelcy/travelcy/services/currency/CurrencyRepository.kt

```diff
@@ -125,25 +125,44 @@ class CurrencyRepository (
             if (response.isSuccessful) {
                 val currencyWebServiceResponse = response.body()
 
-                val currencies = mutableListOf<Currency>()
+                var i = 0
                 currencyWebServiceResponse?.rates?.forEach {
                     val currency = java.util.Currency.getInstance(it.key)
-
-                    currencies.add(
-                        Currency(
+                    val currencyModel: Currency? = currencyDao.getRawCurrency(it.key)
+
+                    if (currencyModel != null) {
+                        currencyModel.name = currency.displayName
+                        currencyModel.exchangeRate = it.value
+                        Log.d(TAG, "Updating currency: ${it.key}")
+                        currencyDao.updateCurrency(currencyModel)
+                    }
+                    else {
+                        Log.d(TAG, "Inserting currency: ${it.key}")
+                        currencyDao.insertCurrency(Currency(
                             it.key,
                             currency.displayName,
-                            it.value
-                        )
-                    )
+                            it.value,
+                            true,
+                            i++
+                        ))
+                    }
                 }
```

Comment:
Shouldn't we clean up currencies that are in the database if they are not retrieved from the update? Else we could end up with stale currencies in the database (for an example if the currency is removed from the web service)

Reply:
Thats a great point! This had been done in 621eda6d74861d239e15c3cef685737a82a5ad24